packages/block-editor/src/store/actions.js

```diff
@@ -1432,36 +1439,128 @@ export const setNavigationMode =
  */
 export const __unstableSetEditorMode =
 	( mode ) =>
-	( { dispatch, select } ) => {
-		// When switching to zoom-out mode, we need to select the root block
-		if ( mode === 'zoom-out' ) {
-			const firstSelectedClientId = select.getBlockSelectionStart();
-			if ( firstSelectedClientId ) {
-				dispatch.selectBlock(
-					select.getBlockHierarchyRootClientId(
+	( { dispatch, select, registry } ) => {
+		registry.batch( () => {
```

Comment:
I figured we may as well just batch everything inside the action. Not really sure why this isn't built-in to how things work by default.